### PR TITLE
feat: dev dot opt-in CTA on io sites

### PR DESCRIPTION
**Pull request by BRKalow**
## ✅ "Ready for Review" checklist

- [ ] The Vercel preview link has been added to this PR's description
- [ ] The Asana task has been added to this PR's description
- [ ] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [ ] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [ ] The description template has been filled in below

---

## 🔗 Relevant links

- [Preview link](https://dev-portal-git-brkfeat-opt-in-io-hashicorp.vercel.app/) 🔎
- [Asana task](url) 🎟️

## 🗒️ What

Adds a CTA for the dev portal beta on Vault and Waypoint's io sites. It's currently behind a flag, which is false for production, so we'll see it in preview deployments but not on the actual io sites yet.

Screen Shot 2022-05-25 at 9 27 19 PM

## 📸 Design Screenshots

Screen Shot 2022-05-25 at 9 24 32 PM

## 🧪 Testing

- [ ] Visit https://dev-portal-git-brkfeat-opt-in-io-hashicorp.vercel.app/
- [ ] Select waypoint or vault in the product switcher
- [ ] Navigate to a docs page
- [ ] Click "Join beta"
- [ ] Validate you are redirect to the correct dev portal route

### CODE REVIEW

File: src/components/_proxied-dot-io/common/dev-dot-opt-in/dev-dot-opt-in.module.css [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  min-width: 16px;
+
+  @media (--medium-up) {
+    align-self: unset;
+    margin-top: 0;
+  }
+}
+
+.alert {
+  margin: 0 !important;
```
Comment (BRKalow) [inline src/components/_proxied-dot-io/common/dev-dot-opt-in/dev-dot-opt-in.module.css:22]: 🔨  this ensures that our `p` styles aren't applied...

File: src/components/_proxied-dot-io/common/dev-dot-opt-in/dev-dot-opt-in.module.css [hunk 2]
```diff
@@ -0,0 +1,38 @@
+.container {
```
Comment (BRKalow) [inline src/components/_proxied-dot-io/common/dev-dot-opt-in/dev-dot-opt-in.module.css:1]: Pretty much copy-pasted this from @kendallstrautman's implementation here: https://github.com/hashicorp/learn/pull/4480

File: src/layouts/_proxied-dot-io/vault/index.tsx [hunk 3]
```diff
@@ -40,39 +43,41 @@ function VaultIoLayout({ children, data }: Props): React.ReactElement {
       />
 
       <Min100Layout footer={<Footer openConsentManager={openConsentManager} />}>
-        {productData.alertBannerActive && (
-          <AlertBanner
-            {...productData.alertBanner}
-            product={productData.slug as Products}
-            hideOnMobile
+        <ProductMetaProvider product={productData.slug as Products}>
```
Comment (BRKalow) [inline src/layouts/_proxied-dot-io/vault/index.tsx:46]: Added the product meta provider so I can access the correct data from the CTA.

File: config/production.json [hunk 4]
```diff
@@ -4,6 +4,7 @@
     "max_static_paths": 9999
   },
   "flags": {
-    "enable_product_docs_search": true
+    "enable_product_docs_search": true,
+    "enable_io_beta_cta": false
```
Comment (BRKalow) [inline config/production.json:8]: New flag, `false` in production so the CTAs won't show up.

Comment (kendallstrautman) [reply config/production.json:8]: Nice, the flag is a great idea 🥇

**Review by kendallstrautman — APPROVED** [review]
I'm unfamiliar with how to access these proxied sites within this preview context, so I wasn't sure how to check out the actual opt-in CTA. But the code looks good!

**Pull request merged**